store/sqlstore/channel_store.go

```diff
@@ -2104,6 +2108,19 @@ func (s SqlChannelStore) UpdateLastViewedAt(channelIds []string, userId string)
 		lastViewedQuery += fmt.Sprintf("WHEN :channelId%d THEN GREATEST(LastViewedAt, :lastViewed%d) ", index, index)
 
 		props["channelId"+strconv.Itoa(index)] = t.Id
+
+		if t.LastPostAt < minLastPostAt {
+			minLastPostAt = t.LastPostAt
```

Comment:
Why the min, and not the max? (The query say GREATEST)

Reply:
@jespino - the query gets the greatest, but of all the selected ones, I am taking the earliest one so that I don't miss anything in the update later on.